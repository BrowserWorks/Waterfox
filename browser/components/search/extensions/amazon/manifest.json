{
  "name": "__MSG_extensionName__",
  "description": "__MSG_extensionDescription__",
  "manifest_version": 2,
  "version": "1.3",
  "applications": {
    "gecko": {
      "id": "amazon@search.waterfox.net"
    }
  },
  "hidden": true,
  "default_locale": "au",
  "icons": {
    "16": "favicon.ico"
  },
  "web_accessible_resources": [
    "favicon.ico"
  ],
  "chrome_settings_overrides": {
    "search_provider": {
      "name": "__MSG_extensionName__",
      "search_url": "__MSG_searchUrl__",
      "search_form": "__MSG_searchForm__",
      "search_url_get_params": "__MSG_searchUrlGetParams__"
    }
  }
}
