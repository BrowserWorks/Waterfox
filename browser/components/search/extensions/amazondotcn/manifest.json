{
  "name": "亚马逊",
  "description": "亚马逊搜索",
  "manifest_version": 2,
  "version": "1.0",
  "applications": {
    "gecko": {
      "id": "amazondotcn@search.waterfox.net"
    }
  },
  "hidden": true,
  "default_locale": "default",
  "icons": {
    "16": "favicon.ico"
  },
  "web_accessible_resources": [
    "favicon.ico"
  ],
  "chrome_settings_overrides": {
    "search_provider": {
      "name": "亚马逊",
      "search_url": "__MSG_searchUrl__",
      "search_form": "https://www.amazon.cn/",
      "search_url_get_params": "__MSG_searchUrlGetParams__"
    }
  }
}
