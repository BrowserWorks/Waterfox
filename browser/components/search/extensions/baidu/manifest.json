{
  "name": "百度",
  "description": "百度网页搜索",
  "manifest_version": 2,
  "version": "1.1",
  "applications": {
    "gecko": {
      "id": "baidu@search.waterfox.net"
    }
  },
  "hidden": true,
  "icons": {
    "16": "favicon.ico"
  },
  "web_accessible_resources": [
    "favicon.ico"
  ],
  "chrome_settings_overrides": {
    "search_provider": {
      "name": "百度",
      "search_url": "https://www.baidu.com/baidu",
      "search_form": "https://www.baidu.com/",
      "search_url_get_params": "tn=monline_7_dg&ie=utf-8&wd={searchTerms}",
      "suggest_url": "https://www.baidu.com/su",
      "suggest_url_get_params": "tn=monline_7_dg&ie=utf-8&action=opensearch&wd={searchTerms}"
    }
  }
}
