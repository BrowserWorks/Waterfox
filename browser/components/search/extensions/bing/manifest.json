{
  "name": "Bing",
  "description": "Microsoft Bing",
  "manifest_version": 2,
  "version": "1.1.1",
  "applications": {
    "gecko": {
      "id": "bing@search.waterfox.net"
    }
  },
  "hidden": true,
  "icons": {
    "16": "favicon.ico"
  },
  "web_accessible_resources": [
    "favicon.ico"
  ],
  "chrome_settings_overrides": {
    "search_provider": {
      "name": "Bing",
      "search_url": "https://www.bing.com/search",
      "search_form": "https://www.bing.com/search?q={searchTerms}&PC=IS45&PTAG=SYS1000000",
      "search_url_get_params": "q={searchTerms}&PC=IS45&PTAG=SYS1000000",
      "params": [
        {
          "name": "form",
          "condition": "purpose",
          "purpose": "contextmenu",
          "value": "ISCHR1"
        },
        {
          "name": "form",
          "condition": "purpose",
          "purpose": "searchbar",
          "value": "ISCHR1"
        },
        {
          "name": "form",
          "condition": "purpose",
          "purpose": "homepage",
          "value": "ISCHR1"
        },
        {
          "name": "form",
          "condition": "purpose",
          "purpose": "keyword",
          "value": "ISCHR1"
        },
        {
          "name": "form",
          "condition": "purpose",
          "purpose": "newtab",
          "value": "ISCHR1"
        }
      ],
      "suggest_url": "https://www.bing.com/osjson.aspx",
      "suggest_url_get_params": "query={searchTerms}&form=OSDJAS&language={moz:locale}"
    }
  }
}
