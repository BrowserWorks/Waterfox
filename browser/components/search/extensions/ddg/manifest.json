{
  "name": "DuckDuckGo",
  "description": "The Internet privacy company that empowers you to seamlessly take control of your personal information online, without any tradeoffs.",
  "manifest_version": 2,
  "version": "1.1",
  "applications": {
    "gecko": {
      "id": "ddg@search.waterfox.net"
    }
  },
  "hidden": true,
  "icons": {
    "16": "favicon.ico"
  },
  "web_accessible_resources": [
    "favicon.ico"
  ],
  "chrome_settings_overrides": {
    "search_provider": {
      "keyword": ["@duckduckgo", "@ddg"],
      "name": "DuckDuckGo",
      "search_url": "https://duckduckgo.com/",
      "search_form": "https://duckduckgo.com/?q={searchTerms}",
      "search_url_get_params": "q={searchTerms}",
      "suggest_url": "https://ac.duckduckgo.com/ac/",
      "suggest_url_get_params": "q={searchTerms}&type=list"
    }
  }
}