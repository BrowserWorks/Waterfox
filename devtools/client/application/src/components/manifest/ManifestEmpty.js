/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { openDocLink } = require("devtools/client/shared/link");

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const {
  a,
  article,
  h1,
  p,
} = require("devtools/client/shared/vendor/react-dom-factories");

const FluentReact = require("devtools/client/shared/vendor/fluent-react");
const Localized = createFactory(FluentReact.Localized);

const DOC_URL =
  "https://developer.mozilla.org/en-US/docs/Web/Manifest" +
  "?utm_source=devtools&utm_medium=sw-panel-blank";

/**
 * This component displays help information when no manifest is found for the
 * current target.
 */
class ManifestEmpty extends PureComponent {
  openDocumentation() {
    openDocLink(DOC_URL);
  }

  render() {
    return article(
      { className: "manifest-empty js-manifest-empty" },
      Localized(
        {
          id: "manifest-empty-intro",
          a: a({
            className: "external-link",
            onClick: () => this.openDocumentation(),
          }),
        },
        h1({ className: "app-page__title" })
      ),
      Localized({ id: "manifest-non-existing" }, p({}))
    );
  }
}

// Exports
module.exports = ManifestEmpty;
