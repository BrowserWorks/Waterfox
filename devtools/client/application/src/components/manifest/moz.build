# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

DevToolsModules(
    'Manifest.js',
	'ManifestColorItem.css',
    'ManifestColorItem.js',
    'ManifestEmpty.js',
    'ManifestIconItem.css',
    'ManifestIconItem.js',
	'ManifestIssue.css',
	'ManifestIssue.js',
	'ManifestIssueList.css',
	'ManifestIssueList.js',
    'ManifestItem.css',
    'ManifestItem.js',
    'ManifestJsonLink.css',
    'ManifestJsonLink.js',
    'ManifestLoader.css',
    'ManifestLoader.js',
    'ManifestPage.js',
    'ManifestSection.css',
    'ManifestSection.js',
)
