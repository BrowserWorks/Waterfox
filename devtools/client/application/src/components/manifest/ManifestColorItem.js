/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const { span } = require("devtools/client/shared/vendor/react-dom-factories");

const Types = require("devtools/client/application/src/types/index");
const ManifestItem = createFactory(
  require("devtools/client/application/src/components/manifest/ManifestItem")
);

/**
 * This component displays a Manifest member which holds a color value
 */
class ManifestColorItem extends PureComponent {
  static get propTypes() {
    return {
      ...Types.manifestItemColor, // { label, value }
    };
  }

  renderColor() {
    const { value } = this.props;
    return value
      ? span(
          {
            className: "manifest-item__color",
            style: { "--color-value": value },
          },
          value
        )
      : null;
  }

  render() {
    const { label } = this.props;
    return ManifestItem({ label }, this.renderColor());
  }
}

module.exports = ManifestColorItem;
