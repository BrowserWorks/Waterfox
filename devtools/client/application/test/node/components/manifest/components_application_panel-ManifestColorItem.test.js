/* Any copyright is dedicated to the Public Domain.
   http://creativecommons.org/publicdomain/zero/1.0/ */

"use strict";

// Import libs
const { shallow } = require("enzyme");
const { createFactory } = require("react");

const ManifestColorItem = createFactory(
  require("devtools/client/application/src/components/manifest/ManifestColorItem")
);

/*
 * Unit tests for the ManifestItem component
 */

describe("ManifestColorItem", () => {
  it("renders the expected snapshot for a populated color item", () => {
    const wrapper = shallow(
      ManifestColorItem({ label: "foo", value: "#ff0000" })
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders the expected snapshot for an empty color item", () => {
    const wrapper = shallow(ManifestColorItem({ label: "foo" }));
    expect(wrapper).toMatchSnapshot();
  });
});
