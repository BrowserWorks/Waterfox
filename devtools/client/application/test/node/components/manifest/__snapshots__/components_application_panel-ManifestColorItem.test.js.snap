// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManifestColorItem renders the expected snapshot for a populated color item 1`] = `
<ManifestItem
  label="foo"
>
  <span
    className="manifest-item__color"
    style={
      Object {
        "--color-value": "#ff0000",
      }
    }
  >
    #ff0000
  </span>
</ManifestItem>
`;

exports[`ManifestColorItem renders the expected snapshot for an empty color item 1`] = `
<ManifestItem
  label="foo"
/>
`;
